### PR TITLE
Enable Unknown GUIDs if requested

### PR DESCRIPTION
**Pull request by Manoplay**
If you have a modded game with custom GUIDs in the game files, especially as stickers, they won't be accepted by the server.
With this modification, you can change this behavior through configuration key  "EnableUnknownGuids".

### CODE REVIEW

File: Refresh.GameServer/Services/GuidCheckerService.cs [hunk 1]
```diff
@@ -43,6 +47,9 @@ private static void ReadStream(Stream stream, HashSet<long> set)
     /// <exception cref="ArgumentOutOfRangeException"></exception>
     public bool IsTextureGuid(TokenGame game, long guid)
     {
+        // Check if all guids are enabled
+        if (this._config.EnableUnknownGuids) return true;
```
Comment (Beyley) [inline Refresh.GameServer/Services/GuidCheckerService.cs:51]: More bikeshedding, but i'd prefer if it was 
```suggestion
        // If specified in the config, allow all GUIDs
        if (this._config.EnableUnknownGuids) return true;
```

File: Refresh.GameServer/Configuration/GameServerConfig.cs [hunk 2]
```diff
@@ -25,4 +25,5 @@ public class GameServerConfig : Config
     public bool RequireGameLoginToRegister { get; set; } = false;
     public bool TrackRequestStatistics { get; set; } = false;
     public string WebExternalUrl { get; set; } = "https://refresh.example.com";
+    public bool EnableUnknownGuids { get; set; } = false;
```
Comment (Beyley) [inline Refresh.GameServer/Configuration/GameServerConfig.cs:28]: A bit of a bikeshed, but i'd prefer the name to be `AllowInvalidTextureGuids`
```suggestion
    public bool AllowInvalidTextureGuids { get; set; } = false;
```

File: Refresh.GameServer/Services/GuidCheckerService.cs [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
 
 public class GuidCheckerService : EndpointService
 {
+    
+    private readonly GameServerConfig _config;
     private readonly HashSet<long> _validMainlineTextureGuids= [];
     private readonly HashSet<long> _validVitaTextureGuids = [];
 
-    internal GuidCheckerService(Logger logger) : base(logger)
+    internal GuidCheckerService( GameServerConfig config, Logger logger) : base(logger)
```
Comment (Beyley) [inline Refresh.GameServer/Services/GuidCheckerService.cs:16]: Small formatting fix
```suggestion
    internal GuidCheckerService(GameServerConfig config, Logger logger) : base(logger)
```

**Review by Beyley — CHANGES_REQUESTED** [review]
Only minor nitpicks on naming and formatting, other than that this is a good change

**Comment by Manoplay** [comment]
I applied all the modifications requested

**Review by Beyley — APPROVED** [review]

**Comment by Beyley** [comment]
Thank you for the contribution!

**Pull request merged**